### ISSUE TITLE
useless-super-delegation when a parameter uses a default value

### ISSUE THREAD

**Issue by PCManticore** [PyCQA/pylint#1085]
Given the following code, pylint will happily emit an useless-super-delegation message. I think it does not make sense that much though, since we can argue that the overridden method is not useless, since it introduces a parameter with a default value.

```
def __init__(self, message='...', **kws):
    super().__init__(message, **kws)
```

**Comment by CMeza99** [comment]
in pylint 1.7.1, it appears to also have an issue with:
```python
class NewObject(SuperObject):
    def __init__(self, name, var):
      super(SuperObject, self).__init__(name, var)
```

**Comment by PCManticore** [comment]
That's an example of why this check exist, you don't have to call the super method if you are not doing anything in your overridden method.

**Comment by CMeza99** [comment]
^^ post updated: I made an error. I am trying to explicitly trying to pass my vars to the parent init.

**Comment by rogalski** [comment]
@CMeza99 once again, how does your code snippet differs from doing:
```python
class NewObject(SuperObject):
    pass
```

**Comment by CMeza99** [comment]
better? I give up.

**Comment by benf-wspdigital** [comment]
> […] the overridden method is not useless, since it introduces a parameter with a default value.

There is also (as described in issue #1547) a false positive when a parameter already has a default, and the override method changes the default.

```python
# lorem.py

class LoremIpsum:

    def odio(self, elit="consecteur"):
        print(elit)


class DolorSitAmet(LoremIpsum):

    def odio(self, elit="adipiscing"):
        super().odio(elit)
```

The check should not match when an override method sets or changes a default for a parameter that is different from the defaults in the superclass method.

**Comment by AWhetter** [comment]
For any beginner contributors looking at this: this is almost identical to #1553 so fixing one may fix the other. But this includes an extra case to consider where we may also want to check whether the default values match, and not just the names of the arguments.

**Comment by OttoWinter** [comment]
Should this be fixed in version 1.8.2? The following code (abbreviated from https://github.com/home-assistant/home-assistant/blob/2ae0c5653ed32c8b7309686d9b897f7d7c6fc0fb/homeassistant/components/cloud/auth_api.py) still produces this error with Python 3.6.4

```python
class PasswordChangeRequired(Exception):
    """Raised when a password change is required."""

    def __init__(self, message='Password change required.'):
        """Initialize a password change required error."""
        super().__init__(message)
```

```bash
$ pylint a.py --disable=all --enable=useless-super-delegation
************* Module a
W:  4, 4: Useless super delegation in method '__init__' (useless-super-delegation)
```

**Comment by hiranya911** [comment]
I'm seeing this with 1.8.4 for the following code:

```
class ExportedUserRecord(UserRecord):
    def __init__(self, data):
        super(ExportedUserRecord, self).__init__(data)

    @property
    def password_hash(self):
        return self._data.get('passwordHash')
```

`self._data` is defined in the parent class, so I have to call the parent constructor.
